docs/fundamentals/networking/sockets/tcp-classes.md

```diff
@@ -78,10 +78,191 @@ The preceding C# code:
+
+```csharp
+var socket = new Socket(SocketType.Stream, ProtocolType.Tcp);
+socket.Connect("www.example.com", 80);
+```
+
+### Create a server socket
+
+In `TcpListener` class, basically, constructors initializes underlying socket. One of them is `TcpListener(IPAddress localaddr, int port)`.
+
+Consider the following TCP listener code:
+
+```csharp
+var listener = new TcpListener(IPAddress.Loopback, 5000);
+```
+
+The preceding TCP listener code is functionally equivalent to the following socket code:
+
+```csharp
+var ep = new IPEndPoint(IPAddress.Loopback, 5000);
+var socket = new Socket(ep.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
+```
+
+### Start listening on the server
+
+The <xref:System.Net.Sockets.TcpListener.Start> method is a wrapper combining Socket's <xref:System.Net.Sockets.Socket.Bind%2A> and <xref:System.Net.Sockets.Socket.Listen>.
+
+Consider the following TCP listener code:
+
+```csharp
+var listener = new TcpListener(IPAddress.Loopback, 5000);
+listener.Start(10);
+```
+
+The preceding TCP listener code is functionally equivalent to the following socket code:
+
+```csharp
+var ep = new IPEndPoint(IPAddress.Loopback, 5000);
+var socket = new Socket(ep.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
+socket.Bind(ep);
+try
+{
+    socket.Listen(10);
+}
+catch (SocketException)
+{
+    socket.Dispose();
+}
+```
+
+### Accepting a connection on the server
+
+`TcpListener` can accept both <xref:System.Net.Sockets.Socket> (via <xref:System.Net.Sockets.TcpListener.AcceptSocket> or <xref:System.Net.Sockets.TcpListener.AcceptSocketAsync>) and <xref:System.Net.Sockets.TcpClient> (via <xref:System.Net.Sockets.TcpListener.AcceptTcpClient> and <xref:System.Net.Sockets.TcpListener.AcceptTcpClientAsync>).
+
+Consider the following TCP client code:
+
+```csharp
+var listener = new TcpListener(IPAddress.Loopback, 5000);
+var acceptSocket = listener.AcceptSocket(); // var acceptSocket = listener.AcceptSocketAsync();
+```
+
+The preceding TCP listener code is functionally equivalent to the following socket code:
+
+```csharp
+var ep = new IPEndPoint(IPAddress.Loopback, 5000);
+var socket = new Socket(ep.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
+var acceptSocket = socket.Accept(); // var acceptSocket = await socket.AcceptAsync();
+```
+
+### Create a `NetworkStream` to send and receive data
+
+`TcpClient` class has a <xref:System.Net.Sockets.TcpClient.GetStream> helper method to instantiate a <xref:System.Net.Sockets.NetworkStream>. With `Socket`, you have to do it manually.
+
+Consider the following `TcpClient` code:
+
+```csharp
+var client = new TcpClient();
+NetworkStream stream = client.GetStream();
+```
+
+Which is equivalent to the following socket code:
+
+```csharp
+var socket = new Socket(SocketType.Stream, ProtocolType.Tcp);
+NetworkStream stream = new(socket, ownsSocket: true); // Please be aware, we're passing true as `ownsSocket` parameter to let stream object to own the socket. 
```

Comment:
We should be more specific, and let's not mix two constructor styles in the same example.

```suggestion
// Be aware that transfering the ownership means that closing/disposing the stream will also close the underlying socket.
var stream = new NetworkStream(socket, ownsSocket: true);
```